webknossos/webknossos/dataset/view.py

```diff
@@ -425,33 +347,55 @@ def write(
             ), f"The number of channels of the dataset ({num_channels}) does not match the number of channels of the passed data ({data.shape[0]})"
 
         mag1_bbox = self._get_mag1_bbox(
-            rel_current_mag_offset=offset,
             rel_mag1_offset=relative_offset,
             abs_mag1_offset=absolute_offset,
             rel_mag1_bbox=relative_bounding_box,
             abs_mag1_bbox=absolute_bounding_box,
             current_mag_size=data_shape,
         )
-        if json_update_allowed:
-            assert self.bounding_box.contains_bbox(
-                mag1_bbox
-            ), f"The bounding box to write {mag1_bbox} is larger than the view's bounding box {self.bounding_box}"
+        assert self.bounding_box.contains_bbox(
+            mag1_bbox
+        ), f"The bounding box to write {mag1_bbox} is larger than the view's bounding box {self.bounding_box}"
 
         current_mag_bbox = mag1_bbox.in_mag(self._mag)
 
+        if not allow_unaligned:
+            if self._data_format == DataFormat.WKW and not self._is_compressed():
+                try:
+                    self._check_shard_alignment(current_mag_bbox)
+                except ValueError:
+                    shard_shape = self.info.shard_shape
+                    warnings.warn(
+                        f"[WARNING] The bounding box to write {current_mag_bbox} is not aligned with the shard shape {shard_shape}. "
+                        "This was supported for uncompressed WKW datasets, but is deprecated now because of concurrency issues. "
+                        "Please use an explicit `allow_unaligned` argument to fix this."
```

Comment:
"Either, ensure that you write shard-aligned chunks OR pass allow_unaligned=True. When using the latter, take care to not write concurrently."